codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -433,6 +433,16 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
             // This method is responsible for ending requests in some situations and must be called
             // when the input has been shutdown.
             super.channelInactive(ctx);
+        } else if (evt instanceof HttpExpectationFailedEvent) {
+            switch (currentState) {
+            case READ_FIXED_LENGTH_CONTENT:
+            case READ_VARIABLE_LENGTH_CONTENT:
+            case READ_CHUNK_SIZE:
+                reset();
+                break;
```

Comment:
I could, but I prefer only to fall though if there is a "logic" justification for it.  This way if someone updates the "default" case they don't necessarily have to worry about other code paths that may not conflict with their updates.

OK to leave as is?